web/app/scripts/black-spots/black-spots-directive.js

```diff
@@ -1,20 +1,90 @@
-(function () {
+(function() {
     'use strict';
 
     /* ngInject */
-    function BlackSpots() {
+    function BlackSpots($q, BoundaryState, LeafletDefaults, InitialState, FilterState,
+        RecordState, TileUrlService, QueryBuilder, BlackspotSets) {
+        var cartoDBAttribution = '&copy; <a href="http://www.openstreetmap.org/copyright">OpenStreetMap</a> contributors, &copy; <a href="http://cartodb.com/attributions">CartoDB</a>';
+        var defaultLayerOptions = {
+            attribution: 'PRS',
+            detectRetina: true
+        };
+
         var module = {
-            restrict: 'EA',
-            templateUrl: 'scripts/black-spots/black-spots-partial.html',
-            bindToController: true,
+            restrict: 'A',
+            scope: false,
             replace: true,
-            controller: 'BlackSpotsController',
-            controllerAs: 'blackspots'
+            controller: '',
+            require: ['leafletMap'],
+            link: link
         };
         return module;
+
+        function link(scope, element, attrs, controllers) {
+            InitialState.ready().then(function() {
+                var leafletController = controllers[0];
+
+                leafletController.getMap().then(addBaseLayers);
+                var maxDateString;
+                if (FilterState.filters.hasOwnProperty('__dateRange')) {
+                    var dtString = FilterState.filters.__dateRange.max;
+                    // If empty, return current time
+                    if (!dtString) {
+                        dtString = new Date().toJSON();
+                        maxDateString = dtString;
+                    }
+                    // If it's already in the right format, don't do the conversion
+                    else if (dtString.indexOf('/') <= 0) {
+                        maxDateString = dtString + 'T23:59:59Z';
+                    } else {
+                        var components = dtString.split('/');
+                        var month = components[0];
+                        var day = components[1];
+                        var year = components[2];
+                        maxDateString = year + '-' + month + '-' + day + 'T23:59:59Z';
+                    }
+                }
+
+                $q.all([leafletController.getMap(),
+                        BlackspotSets.query({
+                            'effective_at': maxDateString
```

Comment:
This needs a record_type parameter as well, so we get the appropriate black spots.